### PR TITLE
logs:made nostr discovery logs more verbose

### LINKED ISSUE #848: [logs]: Make nostr discovery logs more verbose
Clearly show which data is being sent by which nostr relay in each log line.

Use composite log texts like `<some info> | <other info > | <more info>` format.

### PR DESCRIPTION
**Pull request by 0xEgao**
# Pull Request

## Description
Add more details to nostr discovery logs

## Related Issue(s)
Fixes #848 

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Code refactor / performance improvement
- [ ] Documentation update only
- [ ] CI / Docker / Build changes
- [ ] Other (please describe):

## Protocol Version(s) Affected
- [ ] Legacy (ECDSA — `messages.rs`, `contract.rs`, `handlers.rs`)
- [ ] Taproot-Musig2 (`messages2.rs`, `contract2.rs`, `handlers2.rs`)
- [x] Both
- [ ] Neither (infrastructure / tooling only)

> **Note:** When modifying protocol logic, changes must be made to **both** versions unless the
> scope is explicitly version-specific.

## Affected Component(s)
- [ ] **makerd** (background daemon)
- [ ] **taker** (CLI client)
- [ ] **maker-cli** (command-line tool)
- [ ] Core protocol / cryptography / library
- [x] **watch_tower** (contract monitoring & recovery)
- [ ] Docker / deployment scripts
- [ ] Tests / test framework
- [ ] Documentation (`docs/`)
- [ ] Other (please specify):

### Code Quality
- [x] I ran `cargo +nightly fmt --all` and committed the result
- [x] I ran `cargo +stable clippy --all-features --lib --bins --tests -- -D warnings` with zero warnings
- [x] I ran `cargo +stable clippy --examples -- -D warnings` with zero warnings
- [x] I ran `RUSTDOCFLAGS="-D warnings" cargo +nightly doc --all-features --document-private-items --no-deps` with zero warnings
- [x] Pre-commit git hook passes (`ln -s ../../git_hooks/pre-commit .git/hooks/pre-commit` if not already set)

### Testing
- [x] All unit tests pass (`cargo test`)
- [x] Integration tests pass (`cargo test --features integration-test`)
- [x] Changes were manually tested on **regtest**
- [x] End-to-end maker ↔ taker swap tested (where applicable)
- [x] Test-only code is gated behind `#[cfg(feature = "integration-test")]`

### Documentation
- [ ] Relevant files in the `docs/` folder were updated
- [ ] Complex logic is commented

### Security & Privacy (Critical)
- [ ] This change preserves **trustlessness** and **atomic swap guarantees**
- [ ] No regression in **sybil resistance** or **fidelity bond** logic
- [ ] Tor anonymity and P2P message flow were reviewed
- [ ] No new attack vectors or trust assumptions introduced
- [ ] Edge cases and error handling considered
- [ ] ZMQ subscription integrity (raw tx/block feed) is unaffected
- [ ] `integration-test` feature flag is not reachable in production code paths

## Summary by CodeRabbit

* **Chores**
  * Enhanced structured logging and diagnostics across Nostr flows for clearer debug, info, and warning output
  * Improved per-relay visibility for publishes, retries, reads, and responses with contextual fields (relay, event id, payload, errors)
  * More detailed logs for discovery, subscription, event parsing, fidelity validation, caching, and sync progress
  * Heartbeat timing now sourced centrally for background loop consistency

[Review Change Stack](https://app.coderabbit.ai/change-stack/citadel-tech/coinswap/pull/870?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: src/watch_tower/nostr_discovery.rs [hunk 1]
```diff
@@ -270,45 +283,104 @@ fn handle_relay_message(
...
-                            log::info!("Stored verified fidelity via {relay_url}: {txid}:{vout}");
+                            log::info!(
+                                "Stored verified fidelity | relay={} | event_id={} | txid={} | vout={} | maker_address={} | expires_at_height={}",
+                                relay_url,
+                                event.id,
+                                txid,
+                                vout,
+                                maker_address,
+                                expires_at_height
+                            );
```
[the remark this answers is not in the document]
Comment (0xEgao) [reply src/watch_tower/nostr_discovery.rs]: Cannot log directly, as `fidelity` is moved by `insert_fidelity` call.

File: src/nostr_coinswap.rs [hunk 2]
```diff
@@ -165,14 +165,21 @@ pub fn broadcast_bond_on_nostr(
 
     for relay in relays {
         for attempt in 1..=MAX_RETRIES {
+            log::debug!(
```
Comment (mojoX911) [inline src/nostr_coinswap.rs]: lets make this info. Because paylod is already in the debug log above.

**Review by mojoX911 — CHANGES_REQUESTED** [review]
just one nit.

File: src/nostr_coinswap.rs (hunk 2 above)
Comment (0xEgao) [reply src/nostr_coinswap.rs]: Done

**Review by mojoX911 — APPROVED** [review]
Ack
